src/persist/src/indexed/future.rs

```diff
@@ -290,13 +414,144 @@ mod tests {
         f.truncate(Antichain::from_elem(1))?;
 
         // Attempt to read everything in the Future.
-        let snapshot = f.snapshot(Antichain::from_elem(0), None, &blob)?;
-        let updates = snapshot.read_to_end();
+        let updates = slurp_from(&f, &blob, 0, None)?;
 
         assert_eq!(updates, vec![]);
 
         assert_eq!(f.batches.len(), 0);
 
         Ok(())
     }
+
+    #[test]
+    fn future_snapshot() -> Result<(), Error> {
+        let mut blob = BlobCache::new(MemBlob::new("future_snapshot")?);
+        let mut f: BlobFuture<String, String> = BlobFuture::new(BlobFutureMeta {
+            id: Id(0),
+            ts_lower: Antichain::from_elem(0),
+            batches: vec![],
+            next_blob_id: 0,
+        });
+
+        // Construct a batch holding updates for times [3, 5].
+        let updates = vec![
+            (("k".to_string(), "v".to_string()), 3, 1),
+            (("k".to_string(), "v".to_string()), 5, 1),
+        ];
+        let batch = BlobFutureBatch {
+            desc: Description::new(
+                Antichain::from_elem(SeqNo(0)),
+                Antichain::from_elem(SeqNo(2)),
+                Antichain::from_elem(SeqNo(0)),
+            ),
+            updates: updates.clone(),
+        };
+
+        f.append(batch, &mut blob)?;
+
+        let snapshot_updates = slurp_from(&f, &blob, 0, None)?;
+        assert_eq!(updates, snapshot_updates);
+
+        let snapshot_updates = slurp_from(&f, &blob, 0, Some(6))?;
+        assert_eq!(updates, snapshot_updates);
+
+        let snapshot_updates = slurp_from(&f, &blob, 0, Some(2))?;
+        assert!(snapshot_updates.is_empty());
```

Comment:
you'll get a better error message on failure if you assert_eq against an empty vec